plugins/GSdx/res/tfx.fx

```diff
@@ -769,17 +772,22 @@ PS_OUTPUT ps_main(PS_INPUT input)
 		if (C.a < A_one) C.a += A_one;
 	}
 
-#if !SW_BLEND && PS_DITHER
-	ps_dither(C.rgb, input.p.xy);
-	// Dither matrix range is [-4,3] but positive values can cause issues when
-	// software blending is not used or is unavailable.
-	C.rgb -= 3.0;
-#endif
-	
+	if (!SW_BLEND)
+		ps_dither(C.rgb, input.p.xy);
+
 	ps_blend(C, alpha_blend, input.p.xy);
 
 	ps_fbmask(C, input.p.xy);
 
+	// When dithering the bottom 3 bits become meaningless and cause lines in the picture
+	// so we need to limit the color depth on dithered items
+	// SW_BLEND already deals with this so no need to do in those cases
+	if (!SW_BLEND && PS_DITHER && PS_DFMT == FMT_16 && !PS_COLCLIP)
+	{
+		C.rgb = clamp(C.rgb, (float3)0.0f, (float3)255.0f);
```

Comment:
I see this casting pattern like `(float3)0.0f` is applied thoroughly, but what about `0.0f.rgb`  to achieve the same?

Reply:
And what difference would that make? I just copied the code from elsewhere in the shader so it's consistent, I'm not here to overhaul the code

Reply:
No difference at all.
Mine was just an arguable consideration about the style.

Reply:
Fair enough, maybe somebody can address that in the future if they tidy up the shaders, right now I just want the style/code to match the rest of the shader :)